src/zmq/zmqpublishnotifier.cpp

```diff
@@ -75,9 +75,13 @@ bool CZMQAbstractPublishNotifier::Initialize(void *pcontext)
             zmqError("Failed to create socket");
             return false;
         }
-        
-        int HWMlimit = 0;
-        zmq_setsockopt(psocket, ZMQ_SNDHWM, &HWMlimit, sizeof(HWMlimit));
+
+        int hwm = gArgs.GetArg("-zmqhwm", 10000);
+        zmq_setsockopt(psocket, ZMQ_SNDHWM, &hwm, sizeof(hwm));
```

Comment:
The error return should be checked here in case an invalid value is set by the user config.

Reply:
if any invalid value is given, it defaults to1000.